executor/write.go

```diff
@@ -223,21 +229,29 @@ func (e *DeleteExec) deleteMultiTables() error {
 	return nil
 }
 
-func isMatchTableName(entry *RowKeyEntry, tblMap map[int64][]string) bool {
-	names, ok := tblMap[entry.Tbl.Meta().ID]
+func isMatchTableNameNew(tid int64, col *expression.Column, tblMap map[int64][]*ast.TableName) bool {
+	names, ok := tblMap[tid]
 	if !ok {
 		return false
 	}
 	for _, n := range names {
-		if n == entry.TableName {
+		if (col.DBName.L == "" || col.DBName.L == n.Schema.L) && col.TblName.L == n.Name.L {
 			return true
 		}
 	}
-
 	return false
 }
 
 func (e *DeleteExec) deleteSingleTable() error {
+	var (
+		id        int64
+		tbl       table.Table
+		handleCol *expression.Column
+	)
+	for i, t := range e.tblID2Table {
+		id, tbl = i, t
```

Comment:
we can put `handleCol = e.SelectExec.Schema().TblID2handle[id][0]` here and break